### PR TITLE
fix(back-to-top): trigger manage visibility on body height changes

### PR DESCRIPTION
**Pull request by m4olivei**
### Related Ticket(s)

Closes: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/8834

[ADCMS-2857](https://jsw.ibm.com/browse/ADCMS-2857)

### Description

Reproducing this bug was a little tricky within Storybook. What I did was add a story for the Back to Top component called "Default Faux Paged". Initially with the same contents as the "Default" story. At the bottom of the page, there is a button, which when clicked will replace the page contents with other contents, speifically _less_ content, where no Back to Top button would be required at a typical desktop break.

The core of the issue was that, although we have a ResizeObserver to watch changes in the body element height, we don't make use of it to show / hide the Back to Top button when the height changes. I've adjusted the logic and renamed a couple of things, to hopefully make the code more readable.

### Testing instructions

- [ ] Navigate to the Default Faux Paged story under the Back to Top component at a Desktop breakpoint
- [ ] Scroll down the page, until you see the "Next page" button. As you scroll the Back to Top button will show up per usual.
- [ ] Click Next Page. The contents of the page should be replaced.
- [ ] The Back to Top button should dissapear, and not persist. This will depend on your viewport height.
- [ ] Scroll the new content up and down, the Back to Top button should not appear.

### Changelog

**New**

- Back to Top / Default Faux Paged story

**Changed**

- Updated back to top logic to cover body content change scenarios

### CODE REVIEW

**Comment by m4olivei** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by m4olivei** [comment]
recheck

File: packages/web-components/src/components/back-to-top/back-to-top.ts [hunk 1]
```diff
@@ -90,17 +102,28 @@ class DDSBackToTop extends HostListenerMixin(StableSelectorMixin(LitElement)) {
...
 
-  private _observeResizeBody = (entries) => {
+  /**
+   * ResizeObserver callback function.
+   *
+   * We only observe the body element, therefore entries will typically only
+   * have a single item corresponding to changes in the size of the body
+   * element.
+   */
+  private _observeResizeBody = (entries: ResizeObserverEntry[]) => {
```
Comment (andy-blum) [inline packages/web-components/src/components/back-to-top/back-to-top.ts:112]: I think we probably ought to rename this function becuase `observeResizeBody` and `observerResizeBody` is just _begging_ for typos/mis-understandings.

I'd suggest `handleBodyResize` here.

File: packages/web-components/src/components/back-to-top/back-to-top.ts [hunk 2]
```diff
@@ -90,17 +102,28 @@ class DDSBackToTop extends HostListenerMixin(StableSelectorMixin(LitElement)) {
...
-   * Show button only when document height is 3x greater than viewport
+   * The back to top button is only necessary when the document height
+   * is 3x greater than the viewport.
    */
-  private _showBackToTop() {
+  private _isBackToTopNeededForPage() {
     // @ts-ignore
-    this._windowHeight = this.ownerDocument.defaultView?.innerHeight;
-    return this._bodyHeight > this._windowHeight * 3;
+    this._viewportHeight = this.ownerDocument.defaultView?.innerHeight;
```
Comment (andy-blum) [inline packages/web-components/src/components/back-to-top/back-to-top.ts:125]: TIL: [Document.defaultView](https://developer.mozilla.org/en-US/docs/Web/API/Document/defaultView)

**Review by andy-blum — CHANGES_REQUESTED** [review]
Without begin able to actually test this as we wait for the deploy previews to build, I have one change suggestion

File: packages/web-components/src/components/back-to-top/back-to-top.ts (hunk 1 above)
Comment (m4olivei) [reply packages/web-components/src/components/back-to-top/back-to-top.ts]: Good thought. Updated.

File: packages/web-components/src/components/back-to-top/back-to-top.ts (hunk 2 above)
Comment (m4olivei) [reply packages/web-components/src/components/back-to-top/back-to-top.ts:125]: MDN and I were great friends through this, for me also:

* [ownerDocument](https://developer.mozilla.org/en-US/docs/Web/API/Node/ownerDocument)
* [scrollingElement](https://developer.mozilla.org/en-US/docs/Web/API/Document/scrollingElement)

**Review by andy-blum — APPROVED** [review]

**Review by ariellalgilmore — COMMENTED** [review]
Looks great! thanks for the updates. Is it possible though to have the Faux story instead integrated with the default story to include the updates with the next page and then we can remove the additional Faux story?

**Review by ariellalgilmore — APPROVED** [review]
LGTM!

**Comment by m4olivei** [comment]
Thanks for the review @ariellalgilmore !

**Comment by proeung** [comment]
Looks like the Percy React preview needs review, which is blocking the automatic merging. @ariellalgilmore or @annawen1 can you help take a look so that we can get this PR merged?

Screen Shot 2023-01-24 at 11 32 45 AM

**Pull request merged**